odxtools/diaglayercontainer.py

```diff
@@ -43,29 +44,35 @@ def ecus(self) -> NamedItemList[EcuVariant]:
     @staticmethod
     def from_et(et_element: ElementTree.Element, context: OdxDocContext) -> "DiagLayerContainer":
 
-        cat = OdxCategory.category_from_et(et_element, context, doc_type=DocType.CONTAINER)
+        cat = OdxCategory.from_et(et_element, context)
         kwargs = dataclass_fields_asdict(cat)
 
-        protocols = NamedItemList([
-            Protocol.from_et(dl_element, context)
-            for dl_element in et_element.iterfind("PROTOCOLS/PROTOCOL")
-        ])
-        functional_groups = NamedItemList([
-            FunctionalGroup.from_et(dl_element, context)
-            for dl_element in et_element.iterfind("FUNCTIONAL-GROUPS/FUNCTIONAL-GROUP")
-        ])
-        ecu_shared_datas = NamedItemList([
-            EcuSharedData.from_et(dl_element, context)
-            for dl_element in et_element.iterfind("ECU-SHARED-DATAS/ECU-SHARED-DATA")
-        ])
-        base_variants = NamedItemList([
-            BaseVariant.from_et(dl_element, context)
-            for dl_element in et_element.iterfind("BASE-VARIANTS/BASE-VARIANT")
-        ])
-        ecu_variants = NamedItemList([
-            EcuVariant.from_et(dl_element, context)
-            for dl_element in et_element.iterfind("ECU-VARIANTS/ECU-VARIANT")
-        ])
+        def _get_layer_context(_layer_et: ElementTree.Element) -> OdxDocContext:
+            layer_sn = odxrequire(_layer_et.findtext("SHORT-NAME"))
+            layer_docfrag = OdxDocFragment(layer_sn, DocType.LAYER)
+            # add layer doc fragment to container doc fragment
+            return OdxDocContext(context.version, (context.doc_fragments[0], layer_docfrag))
+
+        protocols = NamedItemList[Protocol]()
+        for layer_et in et_element.iterfind("PROTOCOLS/PROTOCOL"):
+            protocols.append(Protocol.from_et(layer_et, _get_layer_context(layer_et)))
```

Comment:
@andlaus does NamedItemList handle appending and internal name lookup update correctly ?

Reply:
this code seems fine to me: the restriction is that after an item has been added to a `NamedItemList`, its `.short_name` attribute must not chang anymore or else accessing the item by its name (`foo_list.bar`) will not be consistent with its `.short_name` anymore...